domain/consensus/processes/pastmediantimemanager/pastmediantimemanager.go

```diff
@@ -46,6 +50,13 @@ func (pmtm *pastMedianTimeManager) PastMedianTime(blockHash *externalapi.DomainH
 	if err != nil {
 		return 0, err
 	}
+	if len(window) == 0 {
+		header, err := pmtm.blockHeaderStore.BlockHeader(pmtm.databaseContext, pmtm.genesisHash)
```

Comment:
It's very weird to check the window size and then give the timestamp of a hard coded hash. I would expect to choose between:
a. if window size is 0 -> return blockHash's timetstamp
b. if blockHash == genesis -> return genesis' timetstamp